gdx/src/com/badlogic/gdx/utils/Queue.java

```diff
@@ -142,8 +141,7 @@ public T removeFirst () {
 
 		final T result = values[head];
 		values[head] = null;
-		head++;
```

Comment:
Personally, I'd refrain from this pattern as it just makes the code harder to read without any performance benefits.